cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -11,36 +11,97 @@
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
 
 import java.util.List;
 
 import static io.strimzi.operator.cluster.model.StorageUtils.parseMemory;
 
 public class Capacity {
-    public static final double DEFAULT_BROKER_DISK_MIB_CAPACITY = 100_000;  // in MiB
-    public static final int DEFAULT_BROKER_CPU_UTILIZATION_CAPACITY = 100;  // as a percentage (0-100)
-    public static final double DEFAULT_BROKER_INBOUND_NETWORK_KIB_PER_SECOND_CAPACITY = 10_000;  // in KiB/s
-    public static final double DEFAULT_BROKER_OUTBOUND_NETWORK_KIB_PER_SECOND_CAPACITY = 10_000;  // in KiB/s
+    private static final int DEFAULT_BROKER_ID = -1;
+    private static final String DEFAULT_BROKER_DOC = "This is the default capacity. Capacity unit used for disk is in MB, cpu is in percentage, network throughput is in KB.";
+
+    private static final double DEFAULT_BROKER_DISK_MIB_CAPACITY = 100_000;  // in MiB
+    private static final int DEFAULT_BROKER_CPU_UTILIZATION_CAPACITY = 100;  // as a percentage (0-100)
+    private static final double DEFAULT_BROKER_INBOUND_NETWORK_KIB_PER_SECOND_CAPACITY = 10_000;  // in KiB/s
+    private static final double DEFAULT_BROKER_OUTBOUND_NETWORK_KIB_PER_SECOND_CAPACITY = 10_000;  // in KiB/s
 
     private Double diskMiB;
     private Integer cpuUtilization;
     private Double inboundNetworkKiBPerSecond;
     private Double outboundNetworkKiBPerSecond;
 
+    private int replicas;
+    private Storage storage;
+
     public Capacity(KafkaSpec spec) {
         BrokerCapacity bc = spec.getCruiseControl().getBrokerCapacity();
 
-        this.diskMiB = bc != null && bc.getDisk() != null ? getSizeInMiB(bc.getDisk()) : generateDiskCapacity(spec.getKafka().getStorage());
+        this.replicas = spec.getKafka().getReplicas();
+        this.storage = spec.getKafka().getStorage();
+
+        this.diskMiB = bc != null && bc.getDisk() != null ? getSizeInMiB(bc.getDisk()) : generateDiskCapacity(storage);
         this.cpuUtilization = bc != null && bc.getCpuUtilization() != null ? bc.getCpuUtilization() : DEFAULT_BROKER_CPU_UTILIZATION_CAPACITY;
         this.inboundNetworkKiBPerSecond = bc != null && bc.getInboundNetwork() != null ? getThroughputInKiB(bc.getInboundNetwork()) : DEFAULT_BROKER_INBOUND_NETWORK_KIB_PER_SECOND_CAPACITY;
         this.outboundNetworkKiBPerSecond = bc != null && bc.getOutboundNetwork() != null ? getThroughputInKiB(bc.getOutboundNetwork()) : DEFAULT_BROKER_OUTBOUND_NETWORK_KIB_PER_SECOND_CAPACITY;
     }
 
     /**
-     * Generate disk capacity configuration from the supplied storage configuration
+     * Generate broker capacity entry for capacity configuration.
+     *
+     * @param brokerId Id of broker
+     * @param diskCapacity Disk capacity configuration
+     * @param doc Documentation for broker entry
+     * @return Broker entry as a JsonObject
+     */
+    private JsonObject generateBrokerCapacity(int brokerId, Object diskCapacity, String doc) {
```

Comment:
I was having trouble with the type here since it was messing up the syntax of the returned JsonObject. I found that the diskCapacity needed to be a:
-  String when there are no JBOD disks
-  JsonObject when there are JBOD
to prevent malformed JSON

Reply:
Can't the String for non-JBOD disks be shared as some JsonObject as well?

Reply:
I need to play around with this a bit more, I'll get back to you

Reply:
Done!